src/main/java/com/imsweb/x12/reader/X12Reader.java

```diff
@@ -277,6 +300,43 @@ else if (loop.getLoop() != null)
         return requiredChildList;
     }
 
+    private boolean checkVersionsAreConsistent(FileType type, Separators separators, Reader reader) throws IOException {
+        if (reader == null || separators == null || type == null)
+            return false;
+
+        char segmentSeparator = separators.getSegment();
+        char elementSeparator = separators.getElement();
+        int c;
+        StringBuilder line = new StringBuilder();
+
+        while ((c = reader.read()) != -1 && c != segmentSeparator)
+            line.append((char)c);
+
+        // The version is the last element
+        // If we got to the end of the file before the end of the line, don't get the version
+        String version = null;
+        if (c == segmentSeparator) {
+            String lineString = line.toString();
+            int versionStartPos = lineString.lastIndexOf(elementSeparator);
+            if (versionStartPos != -1)
+                version = lineString.substring(versionStartPos + 1, lineString.length());
+        }
+        reader.reset();
+
+        boolean result = (FileType.ANSI835_4010_X091.equals(type) && TYPES.get(FileType.ANSI835_4010_X091).equals(version)) ||
+                (FileType.ANSI835_5010_X221.equals(type) && TYPES.get(FileType.ANSI835_5010_X221).equals(version)) ||
+                (FileType.ANSI837_4010_X096.equals(type) && TYPES.get(FileType.ANSI837_4010_X096).equals(version)) ||
+                (FileType.ANSI837_4010_X097.equals(type) && TYPES.get(FileType.ANSI837_4010_X097).equals(version)) ||
+                (FileType.ANSI837_4010_X098.equals(type) && TYPES.get(FileType.ANSI837_4010_X098).equals(version)) ||
+                (FileType.ANSI837_5010_X222.equals(type) && TYPES.get(FileType.ANSI837_5010_X222).equals(version)) ||
+                (FileType.ANSI837_5010_X223.equals(type) && TYPES.get(FileType.ANSI837_5010_X223).equals(version));
```

Comment:
The point of the `Map` was that you don't need this large boolean expression.  You just need to verify that the type in the file is that same as the type in the map.  No need to put specifiec cases.

```java
boolean result = TYPES.get(type).equals(version);
```

That will allow you to just add a new entry to the `Map` to get additional type support.